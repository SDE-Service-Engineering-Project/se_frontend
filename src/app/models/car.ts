export interface Car {
  id: number;
  createdOn: string;
  brand: string;
  model: string;
  constructionYear: number;
  price: number;
  precision: number;
  currency: string;
}
