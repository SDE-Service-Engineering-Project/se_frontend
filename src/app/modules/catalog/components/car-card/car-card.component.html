<div class="card" style="width: 18rem">
  <img
    src="https://pictures.dealer.com/c/costamesacadillac/0679/a976281f56ff06d46dc390283aa14da6x.jpg?impolicy=downsize_bkpt&imdensity=1&w=520"
    class="card-img-top"
    alt="{{ car.brand }} {{ car.model }}"
  />
  <div class="card-body">
    <h5 class="card-title">{{ car.brand }} {{ car.model }}</h5>
    <p class="card-text">Brand: {{ car.brand }}</p>
    <p class="card-text">Model: {{ car.model }}</p>
    <p class="card-text">Manufacture Year: {{ car.constructionYear }}</p>
    <p class="card-text"><b>Price:</b> {{ car.price | currency: car.currency }} / Day</p>
    <button (click)="selectCar()" style="float: right" class="btn btn-primary">
      Rent
    </button>
  </div>
</div>
