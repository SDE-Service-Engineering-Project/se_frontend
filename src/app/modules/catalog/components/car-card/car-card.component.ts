import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Car } from '../../../../models/car';

@Component({
  selector: 'app-car-card',
  templateUrl: './car-card.component.html',
  styleUrls: ['./car-card.component.sass'],
})
export class CarCardComponent {
  @Input() car: Car = {} as Car;
  @Output() carSelected: EventEmitter<Car> = new EventEmitter<Car>();

  selectCar() {
    this.carSelected.emit(this.car);
  }
}
