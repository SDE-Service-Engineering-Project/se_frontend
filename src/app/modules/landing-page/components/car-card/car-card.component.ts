import {Component, EventEmitter, Input, Output} from '@angular/core';
import {Car} from "../../../../models/car";

@Component({
  selector: 'app-car-card',
  templateUrl: './car-card.component.html',
  styleUrls: ['./car-card.component.sass']
})
export class CarCardComponent {
  @Input() car: Car = {} as Car;
  @Output() carRented: EventEmitter<Car> = new EventEmitter<Car>()

  rentCar() {
    this.carRented.emit(this.car);
  }

}
