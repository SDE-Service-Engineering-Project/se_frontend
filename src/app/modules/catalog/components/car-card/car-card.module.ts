import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CarCardComponent } from './car-card.component';

@NgModule({
  imports: [CommonModule],
  declarations: [CarCardComponent],
  exports: [CarCardComponent],
})
export class CarCardModule {}
