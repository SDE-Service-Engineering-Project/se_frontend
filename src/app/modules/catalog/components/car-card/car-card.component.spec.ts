import { ComponentFixture } from '@angular/core/testing';

import { CarCardComponent } from './car-card.component';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { carsMock } from '../../../../utils/testing/mocks/car.mock';

describe('CarCardComponent', () => {
  let component: CarCardComponent;
  let fixture: ComponentFixture<CarCardComponent>;
  let spectator: Spectator<CarCardComponent>;
  let emitSpy: jest.SpyInstance;

  const createComponent = createComponentFactory({
    component: CarCardComponent,
  });

  beforeEach(() => {
    spectator = createComponent();
    fixture = spectator.fixture;
    component = spectator.component;
  });

  it('should render the component', () => {
    expect(fixture.debugElement.nativeElement).toMatchSnapshot();
  });

  it('should emit the car when select car is called', () => {
    emitSpy = jest.spyOn(component.carSelected, 'emit');
    component.car = carsMock[0];
    component.selectCar();
    expect(emitSpy).toHaveBeenCalledWith(carsMock[0]);
  });
});
