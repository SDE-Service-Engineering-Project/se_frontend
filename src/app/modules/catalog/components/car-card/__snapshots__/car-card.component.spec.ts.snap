// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarCardComponent should render the component 1`] = `
<div
  id="root0"
>
  <div
    class="card"
    style="width: 18rem;"
  >
    <img
      alt=" "
      class="card-img-top"
      src="https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Ftse1.mm.bing.net%2Fth%3Fid%3DOIP.GDp9uheqSrMQBmNhBUP1mwHaE0%26pid%3DApi&f=1&ipt=ee353c7a978b7ca256f57cae6ae51c4d5ca321133e806df55516214bfb5ba431&ipo=images"
    />
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
         
      </h5>
      <p
        class="card-text"
      >
        Brand: 
      </p>
      <p
        class="card-text"
      >
        Model: 
      </p>
      <p
        class="card-text"
      >
        Manufacture Year: 
      </p>
      <p
        class="card-text"
      >
        <b>
          Price:
        </b>
          / 
      </p>
      <button
        class="btn btn-primary"
        style="float: right;"
      >
        Rent
      </button>
    </div>
  </div>
</div>
`;
